### DISCUSSION TITLE
create dependency list / tree

### DISCUSSION THREAD

**Discussion by ilka-schulz** [renovatebot/renovate#13150] [Request Help] [answered]
Is there any options to run Renovate without creating merge requests but only listing all dependencies?

And is there a way to let Renovate run recursively on the dependencies? I would love to see a dependency tree of my project. It could be useful to find vulnerabilities in a dependency of a dependency of a dependency...

I do not need any graphical voodoo, plain console output is perfectly fine.

**Comment by rarkins** [comment, marked as answer]
There is not a direct option to do this. But you can run with `--dry-run=true` and then check the `packageFiles with updates` debug log message

**Reply by ilka-schulz** [reply]
Thank you very much! This quick and dirty PoC in Python gives me a dependency list and I will work on it to make it run recursively:

```python
import json

with open(file='renovate-log.ndjson', mode='r') as f:
    lines = f.readlines()
    f.close()

for line in lines:
    obj = json.loads(line)
    if obj["msg"] == "packageFiles with updates":
        # this object holds dependency information
        project = obj['repository']
        print("project: "+project)
        managerlst = obj['config']
        for manager in managerlst:
            # interate over dependencies managed with this manager
            print("    - manager: "+manager)
            deplst = obj['config'][manager][0]['deps']
            for dep in deplst:
                # print dependency
                try:
                    print("        - "+dep['depName'])
                except:
                    try:
                        print("        - "+dep['replaceString'])
                    except:
                        print("        - <unknown>")
```

**Comment by jamietanna** [comment]
As blogged about in https://www.jvt.me/posts/2022/11/01/renovate-dependency-graph/ I've pulled this out with Typescript through https://gitlab.com/tanna.dev/renovate-graph - I'll also raise a feature request to get this officially supported 🤞🏽 

This is used as part of further tooling I've built which can be found linked out from https://dmd.tanna.dev, and includes a command-line tool `dmd` for better querying of the data

**Comment by jamietanna** [comment]
[Feature request](https://github.com/renovatebot/renovate/issues/21455) as mentioned - with some good feedback on it from Rhys 👍

**Comment by jamietanna** [comment]
To anyone coming back to this, also note that [I've written about](https://www.jvt.me/posts/2023/11/03/renovate-to-sbom/) a tool I've created, `renovate-to-sbom`, that takes the data export from Renovate and converts it to one of a few SBOM formats.

That should then be usable by tools that understand SBOMs :+1:
